test/support/with_without_module_doc.ex

```diff
@@ -0,0 +1,66 @@
+defmodule WithModuleDoc do
```

Comment:
I tried different approaches and this seemed to be the easiest way to have the module avaible in the refs and the autlolink tests

Reply:
We don't need to test both autolink and ref test. We already test that autolink warns when something is private, we don't need to add new ones. My suggestion is to move this module to `ref_test` and remove the autolink test.

In a nutshell, autolink should only test the autolink and warning behaviour, it shouldn't really be testing what is public or private, because that's not its responsibility. If we are having a lot of tests around public and private already there, I recommend prunning/simplifying them. :)

Reply:
Got you.
I cannot place the module in the .exs file, because that defeats the purpose of the module which is testing the visibility based on the @doc attributes.
I did my best to simplify the tests in refs_test.exs, please have a look

Reply:
The tests look great, thanks! Do we still need this @moduledoc now with these function refs? :)

Reply:
Fixed. Thanks